### DISCUSSION TITLE
Using TerrainExtension with Google 3D tiles and Mapbox

### DISCUSSION THREAD

**Discussion by joedjc** [visgl/deck.gl#7893] [General]
I'm trying to use TerrainExtension to overlay GeoJSON or MVTs onto Google 3D tiles.

I appreciate this is new and experimental so i'm not sure if this should work and if not whether it may be supported in future.

I'm trying to add MapboxLayers with TerrainExtension but they don't display on the map. 

Here is the code:

`// Add Google 3D tiles as interleaved mapbox layer

map.addLayer(new MapboxLayer({
    id: 'deckgl-3d',
    type: Tile3DLayer,
    data: TILESET_URL,
    onTilesetLoad: tileset3d => {
      tileset3d.options.onTraversalComplete = selectedTiles => {
        const uniqueCredits = new Set();
        selectedTiles.forEach(tile => {
          const {copyright} = tile.content.gltf.asset;
          copyright.split(';').forEach(uniqueCredits.add, uniqueCredits);
        });
        setCredits([...uniqueCredits].join('; '));
        return selectedTiles;
      };
    },
    loadOptions: {
      fetch: {headers: {'X-GOOG-API-KEY': GOOGLE_MAPS_API_KEY}}
    },
    operation: 'terrain+draw'
  }))

// Add a GeoJSON layer using TerrainExtension

map.addLayer(new MapboxLayer({
        id: 'lines',
        type: GeoJsonLayer,
          data: geojson,
          extensions: [new TerrainExtension()],
          stroked: true,
          filled: true,
          getLineColor: [255, 160, 122,255],
          getFillColor: [255, 160, 122,255],
          getLineWidth:50
}))` 

I am also trying to understand if I could drape MVTLayer as well as GeoJSON in the same way.

**Comment by Pessimistress** [comment]
Yes, `TerrainExtension` is supposed to work with both `MVTLayer` and `GeoJsonLayer`. If you need help please create a Codepen/CodeSandbox that reproduces your issue.

**Reply by joedjc** [reply]
Thanks @Pessimistress.

I've set up a codepen below.

https://codepen.io/Joe-djc/pen/dygqbQM

Here is how it looks if I comment out the line "extensions: [new TerrainExtension()]," from the GeoJsonLayer layer:

<img width="1443" alt="Screenshot 2023-05-16 at 17 37 50" src="https://github.com/visgl/deck.gl/assets/13920398/779ace46-d1c5-4b9e-af38-02bf6c62f664">

The line displays but is not draped on the terrain/mesh - it is floating above it presumably at 0 altitude. 

With the terrainExtension enabled the GeoJSON line is not visible on the map. This is using an interleaved mapbox integration approach. If I use the Overlay approach I can get the line to display.

I'm unsure if it's a limitation of the interleaved approach?

**Reply by chrisgervang** [reply]
I'm able to reproduce this draping issue with the terrain extension using the interleaved approach. It doesn't appear to drape.

**Reply by bobfortin64** [reply]
did you try: terrainDrawMode: 'drape' on your geojson layer?

**Reply by LuckDogHu** [reply]
@joedjc I also encountered the same problem, how can I solve this problem

**Reply by taylorwebber** [reply]
@chrisgervang any update on addressing this?

I'm currently running into the same issue.

**Reply by rbraggaar** [reply]
terrainDrawMode: 'drape' does not fix the issue
